### PR TITLE
Added AXI Lite Slave Test

### PR DESCRIPTION
**Pull request by cospan**
Created a test for AXI Lite Slaves.

Included is a core that can interact with the AXI Lite master.
The core will extract the address and data send from the master
and interact with the demo module with a very simple handshake
that the axi_lite_demo module can interface without
a state machine.

The master can read data from axi_lite_demo through the
axi_lite_slave using a simple handshake. The axi_lite_slave
will wrap up a response from the core in an AXI Lite Transaction.

The testbench can be run by navigating to the following directory:

<Cocotb Base>/examples/axi_lite_slave/tests

and running 'make'

Two of the tests fail on purpose, they attempt to read/write
from/to an illegal address.

I did make a change within the directory

<Cocotb Base>/cocotb/drivers/amba.py

where I modified the Exception to be:

AXIProtocolError instead of AXIReadError

Because that exception was raised during writes
and reads

### CODE REVIEW

File: cocotb/drivers/amba.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def read(self, address, sync=True):
             yield RisingEdge(self.clock)
 
         if int(result):
-            raise AXIReadError("Read address 0x%08x failed with RRESP: %d" %
+            raise AXIProtocolError("Read address 0x%08x failed with RRESP: %d" %
                                (address, int(result)))
 
         raise ReturnValue(data)
```
Comment (Martoni) [inline cocotb/drivers/amba.py]: Maybe this should be in a separate commit no ?

File: examples/axi_lite_slave/tests/test_axi_lite_slave.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    DATA = 0xAB
+
+    try:
+        yield axim.write(ADDRESS, DATA)
+        yield Timer(CLK_PERIOD * 10)
+    except AXIProtocolError as e:
+        print "Exception: %s" % str(e)
+        dut.log.info("Bus Successfully Raised an Error")
+        raise TestSuccess()
+    raise TestFaileure("AXI Bus Should have raised an ERROR when writing to \
```
Comment (Martoni) [inline examples/axi_lite_slave/tests/test_axi_lite_slave.py]: s/TestFaileure/TestFailure/ no ?

File: examples/axi_lite_slave/tests/test_axi_lite_slave.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    DATA = 0xAB
+
+    try:
+        yield axim.read(ADDRESS, DATA)
+        yield Timer(CLK_PERIOD * 10)
+    except AXIProtocolError as e:
+        print "Exception: %s" % str(e)
+        dut.log.info("Bus Successfully Raised an Error")
+        raise TestSuccess()
+    raise TestFaileure("AXI Bus Should have raised an ERROR when writing to \
```
Comment (Martoni) [inline examples/axi_lite_slave/tests/test_axi_lite_slave.py]: again s/TestFaileure/TestFailure/ no ?

File: cocotb/drivers/amba.py (hunk 1 above)
Comment (cospan) [inline cocotb/drivers/amba.py]: Yeah it would

File: examples/axi_lite_slave/tests/test_axi_lite_slave.py (hunk 3 above)
Comment (cospan) [inline examples/axi_lite_slave/tests/test_axi_lite_slave.py]: Thanks for the catch

I set the 'expect_error' flag to true so that the test would pass

@cocotb.test(skip = False, expect_error=True)
